### PR TITLE
[Bugfix] Fix Sentry issue: Type Error undefined is not an object

### PR DESCRIPTION
**Pull request by rs-maker**
### Checklist
- [x] I tested *all* changes and *all* related features with different users (student, tutor, editor, instructor, admin) on the test server https://artemistest.ase.in.tum.de.
- [x] Client: I followed the [coding and design guidelines](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/client.html).

### Motivation and Context
 DragAndDropQuestion.mappings is undefined in some cases which leads to a type error when trying to filter the array
 <a href="https://sentry.ase.in.tum.de/organizations/artemis/issues/769/?project=2"> Sentry issue </a>

### Description
 I added a fallback value (true) in case the mappings is not yet defined and also added the '?' operator so some() is not called in this case.

### CODE REVIEW

File: src/main/webapp/app/exercises/quiz/shared/questions/drag-and-drop-question/drag-and-drop-question.component.ts [hunk 1]
```diff
@@ -217,9 +217,11 @@ export class DragAndDropQuestionComponent implements OnChanges {
      */
     getUnassignedDragItems() {
         return this.question.dragItems?.filter((dragItem) => {
-            return !this.mappings.some((mapping) => {
-                return this.dragAndDropQuestionUtil.isSameDragItem(mapping.dragItem!, dragItem);
-            }, this);
+            return (
+                !this.mappings?.some((mapping) => {
+                    return this.dragAndDropQuestionUtil.isSameDragItem(mapping.dragItem!, dragItem);
+                }, this) ?? true
```
Comment (FrankeLukas) [inline src/main/webapp/app/exercises/quiz/shared/questions/drag-and-drop-question/drag-and-drop-question.component.ts:223]: I find
```suggestion
            return (
                !this.mappings || !this.mappings.some((mapping) => {
                    return this.dragAndDropQuestionUtil.isSameDragItem(mapping.dragItem!, dragItem);
                }, this);
```
a bit more readable, but that is just personal preference.

**Review by FrankeLukas — APPROVED** [review]
Changes make sense, thanks.

**Review by FrankeLukas — APPROVED** [review]
Re-approve.

**Review by coolchock — APPROVED** [review]
looks good :+1:

**Pull request merged**